### PR TITLE
Add support for ENUM keys

### PR DESCRIPTION
**Pull request by bhaarathms**
### Issues

Do not know if there is an issue open for this.

### Description

Currently only simple keys are supported. Adding ability to support Enum keys.

### Checklist (Uncheck if it is not completed)

- [ X ] Test cases added
- [  ] Build and test with one-click build and test script passed

### Additional work necessary

No

### CODE REVIEW

**Comment by xuzhg** [comment]
@bhaarathms Would you please help add test cases related to your change?

File: test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ClientEdmModelTests.cs [hunk 1]
```diff
@@ -186,6 +186,15 @@ public void GetOrCreateEdmTypeShouldThrowIfTypeHasObjectProperty()
             test.ShouldThrow<InvalidOperationException>().WithMessage(Strings.ClientType_NoSettableFields("System.Object"));
         }
 
+        [TestMethod]
+        public void GetOrCreateEdmTypeShouldShouldWorkForEnumTypes()
```
Comment (xuzhg) [inline test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ClientEdmModelTests.cs]: double "should"

**Comment by bhaarathms** [comment]
@xuzhg Added tests covering all the changes made in this PR

File: test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ClientEdmModelTests.cs (hunk 1 above)
Comment (bhaarathms) [reply test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ClientEdmModelTests.cs]: fixed

**Pull request merged**